src/util/selinux.c

```diff
@@ -43,25 +44,47 @@
  */
 int selinux_file_context(const char *dst_name)
 {
+    struct selabel_handle *handle = NULL;
     char *scontext = NULL;
+    int ret;
```

Comment:
I think I would split it into two variables: "ret" for storing called functions results and "retval = 1" just for handling this function output.

Reply:
It is safe to return the original errno value (the callers expect 0 on success, non zero on error). I added new patch to address  this.